README.md

```diff
@@ -22,6 +22,196 @@ The XYO protocol for creating origin-blocks is specified in the [XYO Yellow Pape
+
+After creating a genesis block, your origin chain has officially started. Remember, all of the state is stored in the state repository (`XyoOriginChainStateRepository`) and the block repository (`XyoOriginBlockRepository`) that are constructed with the node. Both repositories are very high level and can be implemented for ones needs. Out of the box, this library comes with an implementation for key value store databases (`XyoStorageOriginBlockRepository`) and (`XyoStorageOriginChainStateRepository`). The `XyoStorageProvider` interface defines the methods for a simple key value store. There is a default implementation of an in memory key value store that comes with this library (`XyoInMemoryStorage`).
+
+
+### Creating Origin Blocks
+
+After a node has been created, it can be used to create origin blocks with other nodes. The process of talking to other nodes has been abstracted through use of a pipe (e.g. tcp, ble, memory) that handles all of the transport logic. This interface is defined as `XyoNetworkPipe`. This library ships with a memory pipe, and a tcp pipe.
+
+**Using a tcp pipe** 
+
+```swift
+ // this defines who to create a tcp pipe with
+let tcpPeer = XyoTcpPeer(ip: "myarchivist.com", port: 11000)
+
+// prepares a socket tcp to communicate with the other node
+let socket = XyoTcpSocket.create(peer: tcpPeer)
+
+// wraps the socket to comply to the pipe interface
+let pipe = XyoTcpSocketPipe(socket: socket, initiationData: nil)
+
+// wraps the pipe to preform standard communications
+let handler = XyoNetworkHandler(pipe: pipe)
+
+node.boundWitness(handler: handler, procedureCatalogue: XyoProcedureCatalogue) { (boundWitness, error) in
+    
+}
+```
+
+
+**Using a memory pipe** 
+```swift
+let pipeOne = XyoMemoryPipe()
+let pipeTwo = XyoMemoryPipe()
+
+pipeOne.other = pipeTwo
+pipeTwo.other = pipeOne
+
+let handlerOne = XyoNetworkHandler(pipe: pipeOne)
+let handlerTwo = XyoNetworkHandler(pipe: pipeTwo)
+
+nodeOne.boundWitness(handler: handlerOne, procedureCatalogue: TestInteractionCatalogueCaseOne()) { (result, error) in
+    // this should complete first
+}
+
+nodeTwo.boundWitness(handler: handlerTwo, procedureCatalogue: TestInteractionCatalogueCaseOne()) { (result, error) in
+    // this should complete second
+}
+  ```
+  More example and bridge interactions can be found [here](https://github.com/XYOracleNetwork/sdk-core-swift/tree/docs/sdk-core-swiftTests/node/interaction)
+ 
+ 
+ **Bluetooth**
+ 
+ Bluetooth swift pipes for client and server can be found [here](https://github.com/XYOracleNetwork/sdk-xyobleinterface-swift).
+ 
+ 
+ **Other**
+ 
+ Other network pipes can be implemented as long as the follow the interface defined [here](https://github.com/XYOracleNetwork/sdk-core-swift/blob/docs/sdk-core-swift/network/XyoNetworkPipe.swift).
+ 
+ ### Adding custom data to bound witnesses.
+ 
+ To add custum data to a bound witnesses, a XyoHueresticGetter can be created:
+ 
+ ```swift
+ public struct MyCustomData: XyoHueresticGetter {
```

Comment:
Here, I see that the interface is titled `XyoHueresticGetter`, which should be `XyoHeuristicGetter`?